chalice/app.py

```diff
@@ -626,3 +646,120 @@ def _add_cors_headers(self, response, cors):
         for name, value in cors.get_access_control_headers().items():
             if name not in response.headers:
                 response.headers[name] = value
+
+
+class BuiltinAuthConfig(object):
```

Comment:
There'a already a custom authorizer in the chalice code.  The nomenclature that made the most sense to be (given what's already there):

* Custom authorizers - using the custom authorizer functionality in API gateway with a lambda function that you manage outside of chalice.  This maps to the pre-existing `CustomAuthorizer` class.
* Builtin authorizers - authorizers that are defined in a chalice app that we manage/configure for you.  Implementation wise it uses the custom authorizer feature of API gateway but is largely abstracted in chalice.